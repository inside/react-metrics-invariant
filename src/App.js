import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';
import {metrics} from "react-metrics";

class App extends Component {
  render() {
    return (
      <div className="App">
        <div className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h2>Welcome to React</h2>
        </div>
        <p className="App-intro">
          To get started, edit <code>src/App.js</code> and save to reload.
        </p>
      </div>
    );
  }
}

const config = {
  vendors: [{
    name: "Google Analytics",
    api: {
      pageView: function (eventName, params) {
        // your page view tracking logic here.
      },
      track: function (eventName, params) {
        // your custom link tracking logic here.
      }
    }
  }],
  pageViewEvent: "pageLoad",
  pageDefaults: () => {
    return {
      siteName: "My Web Site"
    };
  }
}

export default metrics(config)(App);
