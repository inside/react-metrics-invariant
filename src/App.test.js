import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

it('renders the snapshot without crashing 1', () => {
  const wrapper = shallow(<App />);
  expect(toJson(wrapper)).toMatchSnapshot();
});

it('renders the snapshot without crashing 2', () => {
  const wrapper = shallow(<App />);
  expect(toJson(wrapper)).toMatchSnapshot();
});
