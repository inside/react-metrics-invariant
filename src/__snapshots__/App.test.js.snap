// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the snapshot without crashing 1 1`] = `
<App
  _metricsConfig={
    Object {
      "attributePrefix": undefined,
      "autoTrackPageView": true,
      "findNewRouteComponent": [Function],
      "getNewRouteState": [Function],
      "suppressTrackBindingWarning": false,
      "useTrackBinding": true,
    }
  }
  metrics={
    Object {
      "pageView": [Function],
      "track": [Function],
    }
  }
/>
`;
